kernel/kernel-impl/src/main/java/org/sakaiproject/messaging/impl/UserMessagingServiceImpl.java

```diff
@@ -101,13 +109,23 @@ public void send() {
 
             // do it
             emailService.sendToUsers(Collections.singleton(user), getHeaders(user.getEmail(), this.subject),
-                    formatMessage(this.subject, this.message));
+                    formatMessage(user.getId(), this.subject, this.message));
 
             log.info("Email sent to: {}", user.getId());
         }
 
         /** helper methods for formatting the message */
-        private String formatMessage(final String subject, final String message) {
+        private String formatMessage(String userId, String subject, String message) {
+
+            try {
+                Site userSite = siteService.getSite(siteService.getUserSiteId(userId));
+                ToolConfiguration tc = userSite.getToolForCommonId("sakai.preferences");
```

Comment:
This has the potential of returning null, if the user's home site doesn't have a Preferences tool. Unlikely, but possible.

I would just change the catch to `Exception` rather than `IdUnusedException`, or multiple catch blocks if you want specific error logging.

Reply:
Good catch Brian, thanks.